plugins/modules/aci_l3out_bfd_interface_profile.py

```diff
@@ -0,0 +1,362 @@
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "absent", ["tenant", "l3out", "l3out_logical_node_profile", "l3out_logical_interface_profile", "bfd_interface_policy"]],
+            ["state", "present", ["tenant", "l3out", "l3out_logical_node_profile", "l3out_logical_interface_profile"]],
+            ["interface_profile_type", "sha1", ["key"]],
+        ],
+    )
+
+    tenant = module.params.get("tenant")
+    l3out = module.params.get("l3out")
+    l3out_logical_node_profile = module.params.get("l3out_logical_node_profile")
+    l3out_logical_interface_profile = module.params.get("l3out_logical_interface_profile")
+    name = module.params.get("name")
+    name_alias = module.params.get("name_alias")
+    description = module.params.get("description")
+    interface_profile_type = module.params.get("interface_profile_type")
+    key = module.params.get("key")
+    key_id = module.params.get("key_id")
+    bfd_interface_policy = module.params.get("bfd_interface_policy")
+    state = module.params.get("state")
+
+    aci = ACIModule(module)
+    aci.construct_url(
+        root_class=dict(
+            aci_class="fvTenant",
+            aci_rn="tn-{0}".format(tenant),
+            module_object=tenant,
+            target_filter={"name": tenant},
+        ),
+        subclass_1=dict(
+            aci_class="l3extOut",
+            aci_rn="out-{0}".format(l3out),
+            module_object=l3out,
+            target_filter={"name": l3out},
+        ),
+        subclass_2=dict(
+            aci_class="l3extLNodeP",
+            aci_rn="lnodep-{0}".format(l3out_logical_node_profile),
+            module_object=l3out_logical_node_profile,
+            target_filter={"name": l3out_logical_node_profile},
+        ),
+        subclass_3=dict(
+            aci_class="l3extLIfP",
+            aci_rn="lifp-{0}".format(l3out_logical_interface_profile),
+            module_object=l3out_logical_interface_profile,
+            target_filter={"name": l3out_logical_interface_profile},
+        ),
+        subclass_4=dict(
+            aci_class="bfdIfP",
+            aci_rn="bfdIfP",
+            module_object="bfdIfP",
+            target_filter={"name": ""},
+        ),
+        child_classes=["bfdRsIfPol"],
+    )
+
+    aci.get_existing()
+
+    if state == "present":
+        child_configs = []
+        class_config = dict(
+            name=name,
+            nameAlias=name_alias,
+            descr=description,
+            key=key,
+            type=interface_profile_type,
+        )
+
+        if key_id and key_id not in range(1, 255):
+            module.fail_json(msg='The "key_id" must be a value between 1 and 255')
+        else:
+            class_config["keyId"] = key_id
+
+        if bfd_interface_policy is not None:
```

Comment:
Is "" enough to delete since the tnXName is pointing to non-existing value, or do we need to delete the object with different payload (status=deleted)?

Reply:
status=deleted will not work in this case- it will throw error here. setting "" will only set the name o the policy argument to empty string and is set to default value by the UI.